pybv/io.py

```diff
@@ -226,30 +224,20 @@ def _write_vmrk_file(vmrk_fname, eeg_fname, events, meas_date):
                   file=fout)
 
 
-def _optimize_channel_unit(resolution, unit):
-    """Calculate an optimal channel scaling factor and unit."""
-    exp = np.log10(resolution)
-    if unit is None:
-        if exp <= -7:
-            return resolution / 1e-9, 'nV'
-        elif exp <= -2:
-            return resolution / 1e-6, 'µV'
-        else:
-            return resolution, 'V'
-    elif unit == 'V':
-        return resolution, 'V'
+def _scale_data_to_unit(data, unit):
+    """Scale `data` in Volts to `data` in `unit`."""
+    if unit == 'V':
+        return data
     elif unit == 'mV':
-        return resolution / 1e-3, 'mV'
+        return data * 1e3
     elif unit in ('µV', 'uV'):
-        return resolution / 1e-6, 'µV'
+        return data * 1e6
     elif unit == 'nV':
-        return resolution / 1e-9, 'nV'
+        return data * 1e9
     else:
         raise ValueError(
             f'Encountered unsupported unit: {unit}'
-            '\nUse either "None" for `unit`, or one of the following: '
-            f'{SUPPORTED_UNITS}'
-            )
+            '\nUse one of the following: {SUPPORTED_UNITS}')
```

Comment:
I believe you'd need to add an `f` here too; and even if it's optional, I'd still add it for visual consistency with the previous line :)

Reply:
agreed, done!

Reply:
Some linters will complain if they find an f-string without any curly braces; if that's the case be sure to `noqa` these lines. I agree that for continued strings it looks better to let all lines be f-strings.